permissions/src/main/java/org/openmbee/sdvc/permissions/config/PermissionInit.java

```diff
@@ -0,0 +1,68 @@
+import org.openmbee.sdvc.rdb.repositories.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.stereotype.Component;
+import javax.transaction.Transactional;
+import java.util.*;
+
+@Component
+@Transactional
+public class PermissionInit implements ApplicationListener<ApplicationReadyEvent> {
+    @Autowired
+    private PrivilegeRepository privRepo;
+    @Autowired
+    private RoleRepository roleRepo;
+
+    @Override
+    public void onApplicationEvent(final ApplicationReadyEvent event) {
+
+        Map<String, List> RPmap = new LinkedHashMap<>();
+        List<String> aPriv = new ArrayList<String>();
+        List<String> rPriv = Arrays.asList(Privileges.values()[0].toString(), Privileges.values()[3].toString(), Privileges.values()[6].toString(), Privileges.values()[8].toString(), Privileges.values()[12].toString(), Privileges.values()[14].toString(), Privileges.values()[18].toString());
+        List<String> wPriv = Arrays.asList(Privileges.values()[0].toString(), Privileges.values()[1].toString(), Privileges.values()[3].toString(), Privileges.values()[4].toString(), Privileges.values()[6].toString(), Privileges.values()[7].toString(), Privileges.values()[8].toString(), Privileges.values()[9].toString(), Privileges.values()[12].toString(), Privileges.values()[13].toString(), Privileges.values()[14].toString(), Privileges.values()[15].toString(), Privileges.values()[18].toString());
+
+        for (int i = 0; i < Privileges.values().length; i++) {
+            aPriv.add(Privileges.values()[i].toString());
+        }
+
+        RPmap.put(Roles.ADMIN.toString(), aPriv);
+        RPmap.put(Roles.READER.toString(), rPriv);
+        RPmap.put(Roles.WRITER.toString(), wPriv);
+
+        for (String role : RPmap.keySet()) {
+            Optional<Role> roleIn = roleRepo.findByName(role);
+            if (!(roleIn.isPresent())) {
+                Role missingRole = new Role();
+                missingRole.setName(role);
+                roleRepo.saveAndFlush(missingRole);
+            }
+        }
+
+        for (String privs : aPriv) {
+            Optional<Privilege> privIn = privRepo.findByName(privs);
+            if (!(privIn.isPresent())) {
+                Privilege missingPriv = new Privilege();
+                missingPriv.setName(privs);
+                privRepo.saveAndFlush(missingPriv);
+            }
+        }
+
+        for (Role role : roleRepo.findAll()) {
+            Set<Privilege> pSet = new HashSet<Privilege>();
+            for (Privilege priv : privRepo.findAll()) {
+                if (((RPmap.get(role.getName())).contains(priv.getName())) && ((role.getPrivileges() == null) || (!(role.getPrivileges().contains(priv))))) {
+                    pSet.add(priv);
+                    role.setPrivileges(pSet);
+                    roleRepo.saveAndFlush(role);
```

Comment:
can move the save outside the priv loop
at the first instance of a priv that's not found, it's resetting the priv set to a set with only 1 priv, then it'll keep adding and saving to the set since the rest of the priv have been removed
just add all the relevant priv to the set and save at the end